syntax = "proto3";

package keto.proto;

/**
 * This is an example of a protobuf message.
 */
message MemoryVaultCreate {
    bytes vault = 1;
    bytes session = 2;
    bytes password = 3;
}

message MemoryVaultAddEntryRequest {
    bytes vault = 1;
    bytes password = 2;
    bytes entry_id = 3;
    bytes value = 4;
}

message MemoryVaultAddEntryResponse {
    bytes entry_id = 1;
    bytes result = 2;
}

message MemoryVaultSetEntryRequest {
    bytes vault = 1;
    bytes password = 2;
    bytes entry_id = 3;
    bytes value = 4;
}

message MemoryVaultSetEntryResponse {
    bytes entry_id = 1;
    bytes result = 2;
}

message MemoryVaultGetEntryRequest {
    bytes vault = 1;
    bytes password = 2;
    bytes entry_id = 3;
}

message MemoryVaultGetEntryResponse {
    bytes entry_id = 1;
    bytes value = 2;
    bytes result = 3;
}


message MemoryVaultRemoveEntryRequest {
    bytes vault = 1;
    bytes password = 2;
    bytes entry_id = 3;
}


message MemoryVaultRemoveEntryResponse {
    bytes entry_id = 1;
    bytes result = 2;
}